Two small fixes

z33ky opened the pull request:
These are two unrelated fixes of issues I discovered.
The commit messages should describe them fine.

---

#### Does this PR close any issues?

<!-- Replace [ ] with [x] for each item your PR satisfies -->
#### PR Checklist
- [x] **My PR follows all guidelines in the CONTRIBUTING.md file**
- [x] My PR targets a `develop` branch OR targets another branch with a specific goal in mind

z33ky commented on sp/src/game/server/env_instructor_hint.cpp:
```diff
@@ -85,8 +99,13 @@ BEGIN_DATADESC( CEnvInstructorHint )
 #ifdef MAPBASE
 	DEFINE_KEYFIELD( m_iszStartSound, FIELD_STRING, "hint_start_sound" ),
 	DEFINE_KEYFIELD( m_iHintTargetPos, FIELD_INTEGER, "hint_target_pos" ),
+	DEFINE_KEYFIELD( m_bActive, FIELD_BOOLEAN, "start_active" ),
```
This field can also be added to the .fgd.

Blixibon commented:
In regards to the `env_instructor_hint` fix, I noticed `m_bActive` doesn't turn off when the hint automatically ends via a cooldown. This means expired hints will continue to appear on save/load for the rest of the map.

It's possible that could be resolved with additional effort, but I'd rather create a more robust clientside save/restore implementation in the future, especially in light of non-`env_instructor_hint` game event lessons, which would never be saved with this code.

The viewmodel zoom fix seems to work fine.

I'd be willing to merge the viewmodel zoom fix in this PR, but I'm not going to merge it with the save/restore fix present unless it is changed to account for hint cooldowns or is removed from the PR.

z33ky commented:
I fixed the timeout.
This is being used as an objective marker in a mod, so restoring on load would be useful. I'm fine carrying the patch in the fork though.
Maybe we'll move to a custom HUD element anyways, where we have more control over the presentation.

Blixibon commented:
For some reason, this code now crashes on this line in `env_instructor_hint.cpp` when the game is restored:
https://github.com/mapbase-source/source-sdk-2013/blob/446ad0935d8206a45968e1aa6c582a9dab2e3b63/sp/src/game/server/env_instructor_hint.cpp#L258

*(That's at least where the stack trace leads, anyway)*

I wasn't able to test this again until now, so I'm really sorry for the delay in getting back to you on this.

z33ky commented:
> For some reason, this code now crashes on this line in env_instructor_hint.cpp when the game is restored

I noticed I switched the operands to calculate `iTimeLeft` in `OnRestore()` and had some supplemental logic errors, which I addressed in the latest commit.
These shouldn't effect whatever error you're seeing though.
Neither the old, nor this new code crash for me. Can you send me the map you're testing on? Note that Hammer doesn't run on Linux, so I won't be able to make one if you just give me a description...

Did you get that stack trace on a release build? I wouldn't really trust it to necessarily get the line right.

> I wasn't able to test this again until now, so I'm really sorry for the delay in getting back to you on this.

Don't sweat it :) If anything I am to blame for introducing bugs :P

Blixibon commented:
The new commit seems to work as intended. I don't see any other problems, so I'll go ahead and merge this now. Thanks for helping resolve these issues.
